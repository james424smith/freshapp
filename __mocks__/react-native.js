import * as ReactNative from "react-native";

export * from "react-native";
export const Vibration = { vibrate: jest.fn() };
export const Adnimated = { spring: jest.fn() };

export default Object.setPrototypeOf(
  {
    Vibration,
    Adnimated,
  },
  ReactNative
);
