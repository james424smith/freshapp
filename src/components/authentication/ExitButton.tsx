import {
  StackActions,
  useNavigation,
  useTheme,
} from "@react-navigation/native";
import React from "react";
import { View } from "react-native";
import { Icon } from "react-native-elements";
import { SIGN_IN_ROUTE, SWIPE_SCREEN_ROUTE } from "../../constants/routes";
import styles from "../../styles/AuthStyles";

interface Props {
  path: "SWIPE" | "SIGN_IN";
  containerStyles?: Record<string, string | number>;
}

const ExitButton = (props: Props) => {
  const theme = useTheme();
  const navigation = useNavigation();

  const dataToUse =
    props.path === "SWIPE"
      ? {
          name: "arrow-forward-circle-outline",
          screen: SWIPE_SCREEN_ROUTE,
        }
      : {
          name: "close-circle-outline",
          screen: SIGN_IN_ROUTE,
        };

  return (
    <View style={props.containerStyles ?? styles.exitButtonContainer}>
      <Icon
        testID={"exit-button"}
        name={dataToUse.name}
        type="ionicon"
        color={theme.colors.blackAndWhite}
        size={30}
        style={styles.skipButton}
        onPress={() =>
          navigation.dispatch(
            StackActions.replace("Auth", {
              screen: dataToUse.screen,
            })
          )
        }
      />
    </View>
  );
};

export default ExitButton;
