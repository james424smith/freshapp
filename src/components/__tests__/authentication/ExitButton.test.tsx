import React from "react";
import ExitButton from "../../authentication/ExitButton";
import { render, cleanup, fireEvent } from "@testing-library/react-native";
import * as RN from "react-native";
import { useNavigation } from "@react-navigation/native";

describe("ExitButton snapshot test", () => {
  afterEach(() => {
    cleanup();
    jest.clearAllMocks();
    jest.restoreAllMocks();
  });

  it("should render correct ui and press the exit icon", () => {
    RN.Platform.OS = "ios";

    const { toJSON, getByTestId } = render(<ExitButton path={"SIGN_IN"} />);

    fireEvent.press(getByTestId("exit-button"));

    expect(useNavigation().dispatch).toHaveBeenCalledTimes(1);
    expect(toJSON()).toMatchSnapshot();
  });
});
