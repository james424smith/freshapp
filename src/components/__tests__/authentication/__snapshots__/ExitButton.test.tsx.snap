// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ExitButton snapshot test should render correct ui and press the exit icon 1`] = `
<View
  style={
    Object {
      "alignItems": "flex-end",
      "paddingTop": 33.5,
    }
  }
>
  <View
    style={
      Object {
        "overflow": "hidden",
      }
    }
  >
    <View
      accessible={true}
      focusable={true}
      onClick={[Function]}
      onResponderGrant={[Function]}
      onResponderMove={[Function]}
      onResponderRelease={[Function]}
      onResponderTerminate={[Function]}
      onResponderTerminationRequest={[Function]}
      onStartShouldSetResponder={[Function]}
      style={
        Object {
          "alignItems": "flex-end",
          "justifyContent": "space-between",
          "padding": 22.5,
        }
      }
      testID="exit-button"
    >
      <View
        style={
          Object {
            "alignItems": "center",
            "backgroundColor": "transparent",
            "justifyContent": "center",
          }
        }
      >
        <Text
          allowFontScaling={false}
          style={
            Array [
              Object {
                "color": "#242424",
                "fontSize": 30,
              },
              Object {
                "backgroundColor": "transparent",
              },
              Object {
                "fontFamily": "Ionicons",
                "fontStyle": "normal",
                "fontWeight": "normal",
              },
              Object {},
            ]
          }
          testID="iconIcon"
        >
          ïˆ¥
        </Text>
      </View>
    </View>
  </View>
</View>
`;
